### PR TITLE
Spawning processor #526

### PR DESCRIPTION
**Pull request by wassoares**
- Created AttendSpawningOrdersThread class with a run method that processes orders by changing the SPAWNING state to FULFILLED or FAILED;
- For the processing were necessary import of the old project: 
  - HttpRequest, SshClientPool and ManagerTimer classes into the utils package;
  - Created CommonConfigurationConstants class with another necessary constants;
  - Create TunnelingServiceUtil class extracting methods responsible for assembling a reverse tunneling address services map;
  - Created SshConnectivityUtil class extracting methods responsible for verifying an active connection of an instance;
  - Created SshCommonUserUtil class extracting method "getSshCommonUser()" common to several classes;
- Created TestAttendSpawningOrdersThread class for testing implementation;

### CODE REVIEW

File: src/main/java/org/fogbowcloud/manager/core/ManagerController.java [hunk 1]
```diff
@@ -1,85 +1,47 @@
...
 
 import org.apache.log4j.Logger;
 import org.fogbowcloud.manager.core.constants.ConfigurationConstants;
-import org.fogbowcloud.manager.core.constants.DefaultConfigurationConstants;
-import org.fogbowcloud.manager.core.datastore.ManagerDatastore;
 import org.fogbowcloud.manager.core.instanceprovider.InstanceProvider;
-import org.fogbowcloud.manager.core.models.orders.Order;
-import org.fogbowcloud.manager.core.models.orders.OrderState;
-import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+import org.fogbowcloud.manager.core.models.orders.OrderRegistry;
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/ManagerController.java]: This class does not exist anymore.

File: src/main/java/org/fogbowcloud/manager/core/ManagerController.java [hunk 2]
```diff
@@ -1,85 +1,47 @@
...
 
-	private ManagerDatastore managerDatastore;
-	private ManagerScheduledExecutorService attendOpenOrdersExecutor;
+	private OrderRegistry orderRegistry;
 
 	private InstanceProvider localInstanceProvider;
 	private InstanceProvider remoteInstanceProvider;
 
-	private static final Logger LOGGER = Logger.getLogger(ManagerController.class);
+	private Thread attendOpenOrdersThread;
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/ManagerController.java]: This thread does not exist.

File: src/main/java/org/fogbowcloud/manager/core/ManagerController.java [hunk 3]
```diff
@@ -1,85 +1,47 @@
...
-					}
-				}
-			}, 0, schedulerPeriod);
-		}
+	public ManagerController(Properties properties, InstanceProvider localInstanceProvider,
+			InstanceProvider remoteInstanceProvider, OrderRegistry orderRegistry) {
+		this.localMemberId = properties.getProperty(ConfigurationConstants.XMPP_ID_KEY);
+		this.localInstanceProvider = localInstanceProvider;
+		this.remoteInstanceProvider = remoteInstanceProvider;
+		this.orderRegistry = orderRegistry;
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/ManagerController.java]: Delete it.

File: src/main/java/org/fogbowcloud/manager/core/ManagerController.java [hunk 4]
```diff
@@ -1,85 +1,47 @@
...
-				}
-			} catch (Exception e) {
-				LOGGER.error("Error while trying to get an Instance for Order: " + System.lineSeparator() + order, e);
-				order.setOrderState(OrderState.FAILED);
-			}
-			this.managerDatastore.updateOrder(order);
-		}
+	private void startManagerThreads() {
+		LOGGER.info("Starting Manager Threads...");
+		this.attendOpenOrdersThread.start();
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/ManagerController.java]: Delete it.

File: src/main/java/org/fogbowcloud/manager/core/constants/DefaultConfigurationConstants.java [hunk 5]
```diff
@@ -3,5 +3,8 @@
 public class DefaultConfigurationConstants {
 	
 	// SCHEDULER PERIODS
-	public static final String OPEN_ORDERS_SCHEDULER_PERIOD = "30000";
+	public static final String OPEN_ORDERS_SLEEP_TIME = "1000"; // 1 second
+	
+	public static final long DEFAULT_INSTANCE_IP_MONITORING_PERIOD = 10000; // reference value is 10 seconds
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/constants/DefaultConfigurationConstants.java]: Please use TimeUnit, is a better practice :smile:

File: src/main/java/org/fogbowcloud/manager/core/models/orders/ComputeOrder.java [hunk 6]
```diff
@@ -69,14 +82,19 @@ public UserData getUserData() {
...
 
+	/**
+	 * These method handle and request an open order, for this, processOpenOrder
+	 * handle the Order to be ready to change your state and request the
+	 * Instance from the InstanceProvider.
+	 */
 	@Override
-	public void handleOpenOrder() {
-		// TODO
+	public synchronized void processOpenOrder(InstanceProvider instanceProvider) {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/models/orders/ComputeOrder.java]: Remove this method.

File: src/main/java/org/fogbowcloud/manager/core/models/orders/NetworkOrder.java [hunk 7]
```diff
@@ -1,55 +1,70 @@
...
+	/**
+	 * These method handle and request an open order, for this, processOpenOrder
+	 * handle the Order to be ready to change your state and request the
+	 * Instance from the InstanceProvider.
+	 */
 	@Override
-	public void handleOpenOrder() {
-		// TODO Auto-generated method stub
-		
+	public synchronized void processOpenOrder(InstanceProvider instanceProvider) {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/models/orders/NetworkOrder.java]: Remove it.

File: src/main/java/org/fogbowcloud/manager/core/models/orders/Order.java [hunk 8]
```diff
@@ -99,24 +112,35 @@ public long getFulfilledTime() {
...
-	
-	public boolean isRemote() {
-		return !this.providingMember.equals(this.requestingMember);
+
+	/**
+	 * These method handle and request an open order, for this, processOpenOrder
+	 * handle the Order to be ready to change your state and request the
+	 * Instance from the InstanceProvider.
+	 */
+	public synchronized void processOpenOrder(InstanceProvider instanceProvider) {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/models/orders/Order.java]: Remove it.

File: src/main/java/org/fogbowcloud/manager/core/models/orders/OrderRegistry.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package org.fogbowcloud.manager.core.models.orders;
+
+public interface OrderRegistry {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/models/orders/OrderRegistry.java]: Delete this Interface.

File: src/main/java/org/fogbowcloud/manager/core/models/orders/StorageOrder.java [hunk 10]
```diff
@@ -26,10 +35,14 @@ public OrderType getType() {
...
+	/**
+	 * These method handle and request an open order, for this, processOpenOrder
+	 * handle the Order to be ready to change your state and request the
+	 * Instance from the InstanceProvider.
+	 */
 	@Override
-	public void handleOpenOrder() {
-		// TODO Auto-generated method stub
-		
+	public synchronized void processOpenOrder(InstanceProvider instanceProvider) {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/models/orders/StorageOrder.java]: Remove this method.

File: src/main/java/org/fogbowcloud/manager/core/threads/AttendOpenOrdersThread.java [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.constants.ConfigurationConstants;
+import org.fogbowcloud.manager.core.constants.DefaultConfigurationConstants;
+import org.fogbowcloud.manager.core.instanceprovider.InstanceProvider;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderRegistry;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+
+public class AttendOpenOrdersThread extends Thread {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/threads/AttendOpenOrdersThread.java]: Remove this class

File: src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.orders.OrderType;
+import org.fogbowcloud.manager.core.models.orders.instances.ComputeOrderInstance;
+import org.fogbowcloud.manager.core.models.orders.instances.InstanceState;
+import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+import org.fogbowcloud.manager.core.plugins.ComputePlugin;
+import org.fogbowcloud.manager.core.utils.SshConnectivityUtil;
+import org.fogbowcloud.manager.core.utils.TunnelingServiceUtil;
+
+public class AttendSpawningOrdersThread extends Thread {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java]: I do not think that the name of this Thread should be AttendSpawningOrdersThread, check the document contract. I think that is SpawningProcessor. Therefore, check this [PR](https://github.com/fogbow/fogbow-manager-core/pull/13/files) with the OpenProcessor Thread implementation revised.

File: src/main/java/org/fogbowcloud/manager/core/utils/ManagerTimerUtil.java [hunk 13]
```diff
@@ -0,0 +1,44 @@
+package org.fogbowcloud.manager.core.utils;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+
+public class ManagerTimerUtil {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/utils/ManagerTimerUtil.java]: Delete this class.

File: src/test/java/org/fogbowcloud/manager/core/threads/TestAttendOpenOrdersThread.java [hunk 14]
```diff
@@ -0,0 +1,327 @@
...
+import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+import org.fogbowcloud.manager.core.models.token.Token;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+public class TestAttendOpenOrdersThread {
```
Comment (paulofelipefeitosa) [inline src/test/java/org/fogbowcloud/manager/core/threads/TestAttendOpenOrdersThread.java]: Delete this class.

File: src/main/java/org/fogbowcloud/manager/core/utils/TunnelingServiceUtil.java [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+					servicePerAddress.put(service, sshPublicHostIP + ":" + port);
+				}
+				return servicePerAddress;
+			}
+		} catch (Throwable e) {
+			LOGGER.error("Error trying to communicate reverse tunnel or set map of addresses (IP and Port) for order [" + orderId + "]", e);
+		} finally {
+			if (response != null) {
+				try {
+					response.getEntity().getContent().close();
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/utils/TunnelingServiceUtil.java]: Methods call chaining.

File: src/main/java/org/fogbowcloud/manager/core/utils/TunnelingServiceUtil.java [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+				JSONObject jsonPorts = new JSONObject(EntityUtils.toString(response.getEntity()));
+				if (jsonPorts.isNull(CommonConfigurationConstants.SSH_SERVICE_NAME)) {
+					return null;
+				}
+				@SuppressWarnings("unchecked")
+				Iterator<String> serviceIterator = jsonPorts.keys();
+				Map<String, String> servicePerAddress = new HashMap<String, String>();
+				String sshPublicHostIP = this.properties.getProperty(ConfigurationConstants.TOKEN_HOST_PUBLIC_ADDRESS_KEY);
+				while (serviceIterator.hasNext()) {
+					String service = (String) serviceIterator.next();
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/utils/TunnelingServiceUtil.java]: This cast is not necessary.

File: src/main/java/org/fogbowcloud/manager/core/utils/TunnelingServiceUtil.java [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+					String port = jsonPorts.optString(service);
+					servicePerAddress.put(service, sshPublicHostIP + ":" + port);
+				}
+				return servicePerAddress;
+			}
+		} catch (Throwable e) {
+			LOGGER.error("Error trying to communicate reverse tunnel or set map of addresses (IP and Port) for order [" + orderId + "]", e);
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/utils/TunnelingServiceUtil.java]: *_Error while_

File: src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java [hunk 18]
```diff
@@ -0,0 +1,224 @@
...
+		Assert.assertNull(SharedOrderHolders.getInstance().getSpawningOrdersList().getNext());
+		
+		Order test = SharedOrderHolders.getInstance().getFulfilledOrdersList().getNext();
+		assertNotNull(test);
+		Assert.assertEquals(order.getOrderInstance(), test.getOrderInstance());
+		Assert.assertEquals(OrderState.FULFILLED, test.getOrderState());
+	}
+	
+	@Test
+	public void testGetExternalServiceAddressesConfiguredToThrowException() {
```
Comment (paulofelipefeitosa) [inline src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java]: I do not understand this test main goal, by the fact that after calling the method processSpawningOrder there is no end checks. In this test case the Order will continue in the Spawning state right?

File: src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java [hunk 19]
```diff
@@ -0,0 +1,224 @@
...
+		OrderInstance orderInstance = new OrderInstance(DEFAULT_ORDER_INSTANCE_ID);
+		order.setOrderInstance(orderInstance);
+		
+		ComputeOrderInstance computeOrderInstance = Mockito.spy(ComputeOrderInstance.class);
+		computeOrderInstance.setState(InstanceState.ACTIVE);
+		Mockito.when(this.computePlugin.getInstance(Mockito.any(Token.class), Mockito.eq(DEFAULT_ORDER_INSTANCE_ID))).thenReturn(computeOrderInstance);
+		
+		TunnelingServiceUtil tunnelingService = Mockito.mock(TunnelingServiceUtil.class);
+		this.attendSpawningOrdersThread.setTunnelingService(tunnelingService);
+		when(tunnelingService.getExternalServiceAddresses(order.getId())).thenReturn(null);
```
Comment (paulofelipefeitosa) [inline src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java]: I do not think that this line is necessary.

File: src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java [hunk 20]
```diff
@@ -0,0 +1,224 @@
...
+		Assert.assertEquals(OrderState.FAILED, test.getOrderState());
+	}
+	
+	@Test 
+	public void testProcesseComputeOrderInstanceConfiguredToThrowException() {
+		Order order = this.createOrder();
+		order.setOrderState(OrderState.SPAWNING);
+		OrderInstance orderInstance = order.getOrderInstance();
+		order.setOrderInstance(orderInstance);
+		when(this.computePlugin.getInstance(Mockito.any(Token.class), Mockito.eq(DEFAULT_ORDER_INSTANCE_ID))).thenReturn(null);
```
Comment (paulofelipefeitosa) [inline src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java]: I do not think that this line is necessary.

File: src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java [hunk 21]
```diff
@@ -0,0 +1,224 @@
...
+		this.attendSpawningOrdersThread.processSpawningOrder(order);
+		Assert.assertNull(SharedOrderHolders.getInstance().getSpawningOrdersList().getNext());
+		
+		Order test = SharedOrderHolders.getInstance().getFailedOrdersList().getNext();		
+		Assert.assertEquals(order.getOrderInstance(), test.getOrderInstance());
+		Assert.assertEquals(OrderState.FAILED, test.getOrderState());
+	}
+	
+	@Test 
+	public void testProcesseComputeOrderInstanceConfiguredToThrowException() {
```
Comment (paulofelipefeitosa) [inline src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java]: We need to discuss about this test case and the implementation impact.

File: src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java [hunk 22]
```diff
@@ -0,0 +1,224 @@
...
+		order.setOrderState(OrderState.SPAWNING);
+		OrderInstance orderInstance = order.getOrderInstance();
+		order.setOrderInstance(orderInstance);
+		when(this.computePlugin.getInstance(Mockito.any(Token.class), Mockito.eq(DEFAULT_ORDER_INSTANCE_ID))).thenReturn(null);
+		when(this.computePlugin.getInstance(Mockito.any(Token.class), Mockito.eq(DEFAULT_ORDER_INSTANCE_ID))).thenThrow(new RuntimeException("Any Exception"));
+		this.attendSpawningOrdersThread.processSpawningOrder(order);
+	}
+	
+	@Test
+	public void  testMethodProcessSpawningOrderConfiguredToThrowException() {
```
Comment (paulofelipefeitosa) [inline src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java]: What is the main goal of this test case? There is any chance of the getOrderInstance throw an Exception?

File: src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+				order = SharedOrderHolders.getInstance().getSpawningOrdersList().getNext();
+				if (order != null) {
+					this.processSpawningOrder(order);
+				} else {
+					LOGGER.info("There is no spawning order to be processed, sleeping attend spawning orders thread.");
+					Thread.sleep(this.sleepTime);
+				}
+			} catch (Throwable e) {
+				String orderId = order != null ? order.getId() : null;
+				LOGGER.error("Error while trying to thread sleep attending the order: [" + orderId + "]", e);
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java]: I think that is not necessary to log the Order here.

File: src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+			}
+		}
+		
+	}
+
+	private void setTunnelingServiceAddresses(Order order, ComputeOrderInstance computeOrderInstance) {
+		synchronized (order) {
+			try {
+				Map<String, String> externalServiceAddresses = tunnelingService.getExternalServiceAddresses(order.getId());
+				if (externalServiceAddresses != null) {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java]: Apparently your JUnit tests for this class do not cover the case that the externalServiceAddresses is null.

File: src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java [hunk 25]
```diff
@@ -0,0 +1,129 @@
...
+				LOGGER.error("Error while trying to thread sleep attending the order: [" + orderId + "]", e);
+			}
+		}
+	}
+
+	protected void processSpawningOrder(Order order) {
+		synchronized (order) {
+			OrderState orderState = order.getOrderState();
+			if (orderState.equals(OrderState.SPAWNING)) {
+				LOGGER.info("Trying to get an instance for order [" + order.getId() + "]");
```
[the remark this answers is not in the document]
Comment (wassoares) [reply src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java]: Remove the logger and try catch, or rectify only the info mensage?

File: src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java (hunk 12 above)
Comment (wassoares) [reply src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java]: Thanks for the tip, of according to the document contract, this is must be  SpawningMonitor...

File: src/main/java/org/fogbowcloud/manager/core/models/orders/StorageOrder.java (hunk 10 above)
Comment (wassoares) [reply src/main/java/org/fogbowcloud/manager/core/models/orders/StorageOrder.java]: I don't have this method in my branch.

File: src/main/java/org/fogbowcloud/manager/core/ManagerController.java (hunk 1 above)
Comment (wassoares) [reply src/main/java/org/fogbowcloud/manager/core/ManagerController.java]: This classes in reds lines don't exist in my branch, only the one identified in the green line. 
The codes in reds lines might belong to the branch create-compute.

File: src/main/java/org/fogbowcloud/manager/core/models/orders/Order.java [hunk 26]
```diff
@@ -22,18 +25,28 @@
...
+	public Order(String id, Token localToken, Token federationToken, String requestingMember, String providingMember) {
+		this.id = id;
+		this.orderState = OrderState.OPEN;
+		this.localToken = localToken;
+		this.federationToken = federationToken;
+		this.requestingMember = requestingMember;
+		this.providingMember = providingMember;
+	}
+
+	public Order(Token localToken, Token federationToken, String requestingMember, String providingMember) {
```
Comment (llvieira) [inline src/main/java/org/fogbowcloud/manager/core/models/orders/Order.java]: [Just an improvement] For this constructor you can reuse the first Order constructor. You can do: 

```java
public Order(Token localToken, Token federationToken, String requestingMember, String providingMember) {
        String id = UUID.randomUUID().toString();
	this(id, localToken, federationToken, requestingMember, providingMember);
}
```

**Comment by thiagomanel** [comment]
@wassoares it seems you've pushed some binary files (.class). If so, please remove

**Comment by thiagomanel** [comment]
@wassoares same comment as before, now for the bin/ directory. please, review the whole dataset to check other files that should not be in the repository

File: src/main/java/org/fogbowcloud/manager/core/constants/CommonConfigurationConstants.java [hunk 27]
```diff
@@ -0,0 +1,26 @@
+package org.fogbowcloud.manager.core.constants;
+
+public class CommonConfigurationConstants {
+
+	// from the old class ManagerController in fogbow-manager
+	public static final String DEFAULT_COMMON_SSH_USER = "fogbow";
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/constants/CommonConfigurationConstants.java]: If _SSH_COMMON_USER_KEY_ is a configuration constant, then your default value should be in the file **DefaultConfigurationConstants**.

File: src/main/java/org/fogbowcloud/manager/core/threads/SpawningMonitor.java [hunk 28]
```diff
@@ -0,0 +1,142 @@
...
+
+	/**
+	 * This method does not synchronize the order object because it is private and
+	 * can only be called by the processSpawningOrder method.
+	 */
+	private void processInstance(Order order) throws OrderStateTransitionException {
+
+		OrderInstance orderInstance = order.getOrderInstance();
+
+		if (order.getType().equals(OrderType.COMPUTE)) {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/threads/SpawningMonitor.java]: Method call chaining.

File: src/main/java/org/fogbowcloud/manager/core/threads/SpawningMonitor.java [hunk 29]
```diff
@@ -0,0 +1,142 @@
...
+			if (orderInstance.getState().equals(InstanceState.FAILED)) {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/threads/SpawningMonitor.java]: Method call chaining.

File: src/main/java/org/fogbowcloud/manager/core/threads/SpawningMonitor.java [hunk 30]
```diff
@@ -0,0 +1,142 @@
...
+				LOGGER.info("The compute instance state is failed for order [" + order.getId() + "]");
+				OrderStateTransitioner.transition(order, OrderState.FAILED);
+
+			} else if (orderInstance.getState().equals(InstanceState.ACTIVE)) {
```
Comment (paulofelipefeitosa) [inline src/main/java/org/fogbowcloud/manager/core/threads/SpawningMonitor.java]: Method call chaining.

**Pull request merged**